### PR TITLE
AWS EC2 tweaks for Debian images

### PR DESCRIPTION
**Pull request by msheiny**
Nothing crazy, just noticed that the submenu text was coming up different and I
added a config that I was able to compile against. Note the `awsfull` designator
thats because this config includes all the bloat that I considered included with
the default AMI config.

### CODE REVIEW

File: roles/install-grsec-kernel/tasks/grub_config.yml [hunk 1]
```diff
@@ -25,6 +25,17 @@
       entry exists in the GRUB config file.
   when: grub_grsecurity_menu_entry is not defined
 
+- name: Check whether GRUB DISTRIBUTOR is defined
+  command: grep -q GRUB_DISTRIBUTOR /etc/default/grub
+  register: reg_grub_distributor_check
+  ignore_errors: yes
+  when: ansible_distribution == "Debian"
```
Comment (conorsch) [inline roles/install-grsec-kernel/tasks/grub_config.yml]: This'll need a `changed_when: false`, since it's a read-only task that will never change state.

**Comment by conorsch** [comment]
Great commit comments, particularly f8eeab9. If I'm following that logic right, we should prepend `Debian ` to the grub submenu string if-and-only-if GRUB_DISTRIBUTOR is set in /etc/default/grub. Seems like a saner approach would be to read that value, but we can't rely on it being a string, as on some hosts it's enclosed in backticks, and defaults to "Debian".

Logic looks sound, I'll have to take this for a spin and try it out!

**Comment by conorsch** [comment]
Rebased on top of latest master.

**Comment by conorsch** [comment]
Functionally this is sound: I was able to build an AWS AMI running a grsecurity-patched kernel based off of this PR. Have a few opinions about style, so will add some revisions here prior to merge.

**Comment by conorsch** [comment]
Cleaned up some stylistic nits. Taking a step back to evaluate whether the GRUB_DISTRIBUTOR check is the sanest way to determine the prefix. Looks like the submenu-text-deriving logic should be folded into the custom facts module and set there for all hosts. Will take a look and test on a few different backend providers to make sure it's working across the board.

**Comment by msheiny** [comment]
>Taking a step back to evaluate whether the GRUB_DISTRIBUTOR check is the sanest way to determine the prefix.

Yay thats a good call. I made the detection decision very hastily to get things working. Havent tested across different distros besides debian locally, in AWS and DigitalOcean .

**Comment by msheiny** [comment]
rebased ontop of `master`

**Comment by conorsch** [comment]
In the context of #109, we should get this in before splitting up the roles. Rebased the feature branch on latest master, and cleaned up the commits a bit. Confirmed that the changes do not affect installation on Ubuntu hosts in Vagrant VMs. Did not double-check against an AWS EC2 instance, but we can circle back on any problems once the install role has a dedicated repository.

**Pull request merged**